data-prepper-plugins/http-source/src/main/java/org/opensearch/dataprepper/plugins/source/loghttp/HTTPSource.java

```diff
@@ -49,7 +50,8 @@ public class HTTPSource implements Source<Record<Log>> {
     private static final String HTTP_HEALTH_CHECK_PATH = "/health";
 
     @DataPrepperPluginConstructor
-    public HTTPSource(final HTTPSourceConfig sourceConfig, final PluginMetrics pluginMetrics, final PluginFactory pluginFactory) {
+    public HTTPSource(final HTTPSourceConfig sourceConfig, final PluginMetrics pluginMetrics, final PluginFactory pluginFactory,
+                      final PipelineDescription pipelineDescription) {
```

Comment:
Can I get some context on why `pipelineDescription` was added?

Reply:
On line 73, it provides the pipelineName to the plugin settings.